dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/dataitems/DataItemQueryTests.java

```diff
@@ -0,0 +1,297 @@
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.is;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.hisp.dhis.ApiTest;
+import org.hisp.dhis.actions.LoginActions;
+import org.hisp.dhis.actions.dataitem.DataItemActions;
+import org.hisp.dhis.dto.ApiResponse;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test cases related to GET "dataItems" endpoint. The tests and assertions are
+ * based on the file "setup/metadata.json" => "programIndicators",
+ * "dataElements".
+ * 
+ * The test cases using default pagination will imply "paging=true", which is
+ * the default when "paging" is omitted.
+ *
+ * @author maikel arabori
+ */
+public class DataItemQueryTests extends ApiTest
+{
+    private static final int FOUND = 302;
+
+    private static final int NOT_FOUND = 404;
+
+    private static final int CONFLICT = 409;
+
+    private DataItemActions dataItemActions;
+
+    @BeforeAll
+    public void before()
+    {
+        dataItemActions = new DataItemActions();
+
+        login();
+    }
+
+    @Test
+    public void testGetAllDataItemsUsingDefaultPagination()
+    {
+        // When
+        final ApiResponse response = dataItemActions.get();
+
+        // Then
+        response.validate().statusCode( is( FOUND ) );
+        response.validate().body( "pager", isA( Object.class ) );
+        response.validate().body( "nameableObjects", isA( Object.class ) );
+        response.validate().body( "nameableObjects", is( not( empty() ) ) );
+        assertThat( hasMultipleDataItemTypes( response ), is( true ) );
+    }
+
+    @Test
+    public void testGetAllDataItemsWithoutPagination()
+    {
+        // Given
+        final String noPagination = "?paging=false";
+
+        // When
+        final ApiResponse response = dataItemActions.get( noPagination );
+
+        // Then
+        response.validate().statusCode( is( FOUND ) );
+        response.validate().body( "pager", is( nullValue() ) );
+        response.validate().body( "nameableObjects", isA( Object.class ) );
+        response.validate().body( "nameableObjects", is( not( empty() ) ) );
+        assertThat( hasMultipleDataItemTypes( response ), is( true ) );
+    }
+
+    @Test
+    public void testGetAllDataItemsUsingDefaultPaginationOrderedByCode()
+    {
+        // When
+        final ApiResponse response = dataItemActions.get( "?order=code:asc" );
+
+        // Then
+        response.validate().statusCode( is( FOUND ) );
+        response.validate().body( "pager", isA( Object.class ) );
+        response.validate().body( "nameableObjects", isA( Object.class ) );
+        response.validate().body( "nameableObjects[0].dimensionItemType", isA( String.class ) );
+        response.validate().body( "nameableObjects[0].code", is( "AAAAAAA-1234" ) );
```

Comment:
Do you import this data item before the tests? We run them on empty db.

Reply:
Yes, I added a new "programIndicators" attribute to the "metadata.json". It should be imported during the execution chain. Or not?

Reply:
yep, that's correct!